### ISSUE TITLE
Yocto build error

### ISSUE THREAD

**Issue by jwinarske** [jerel/membrane#10]
Moving to tip of tree I'm hitting this build error:

```
|      Running `rustc --crate-name serde_bytes --edition=2018 /__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0/cargo_home/bitbake/serde_bytes-0.11.5/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=53606f82d515ffee -C extra-filename=-53606f82d515ffee --out-dir /__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0/build/target/x86_64-poky-linux/release/deps --target x86_64-poky-linux -C linker=/__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0/wrapper/target-rust-ccld -L dependency=/__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0/build/target/x86_64-poky-linux/release/deps -L dependency=/__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0/build/target/release/deps --extern serde=/__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0/build/target/x86_64-poky-linux/release/deps/libserde-8c571d51b5658631.rmeta --cap-lints allow -L /__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0/recipe-sysroot/usr/lib/rust --remap-path-prefix=/__w/meta-flutter/yocto-kirkstone/build/tmp/work/core2-64-poky-linux/membrane-dart-example/git-r0=/usr/src/debug/membrane-dart-example/git-r0`
[1181](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1182)
| error[E0107]: this struct takes 3 generic arguments but 2 generic arguments were supplied
[1182](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1183)
|   --> /usr/src/debug/membrane-dart-example/git-r0/cargo_home/bitbake/serde_yaml-0.8.21/src/mapping.rs:16:10
[1183](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1184)
|    |
[1184](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1185)
| 16 |     map: IndexMap<Value, Value>,
[1185](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1186)
|    |          ^^^^^^^^ -----  ----- supplied 2 generic arguments
[1186](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1187)
|    |          |
[1187](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1188)
|    |          expected 3 generic arguments
[1188](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1189)
|    |
[1189](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1190)
| help: add missing generic argument
[1190](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1191)
|    |
[1191](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1192)
| 16 |     map: IndexMap<Value, Value, S>,
[1192](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1193)
|    |                               +++
[1193](https://github.com/meta-flutter/meta-flutter/runs/6757718596?check_suite_focus=true#step:9:1194)
|
```

**Comment by jerel** [comment]
Thanks for the report and error log. This appears to have been a configuration edge case in `serde_yaml` but [has been fixed in version 0.8.24](https://github.com/dtolnay/serde-yaml/pull/243). My recent changes on the `main` branch that could affect builds is:
1. Routine updates of several dependencies to newer patch versions (no serde packages were updated).
1. Removal of the Cargo.lock file since this project is used by other projects as a library. The lockfiles for `lib` projects are ignored by Cargo anyway so I wouldn't expect this to have any effect unless Yocto was somehow relying on it.

I notice one peculiarity... the version shown [in the bitbake config](https://github.com/meta-flutter/meta-flutter/blob/3cd0ace6e85357fd41097a5878271097cb3eedaa/recipes-devtools/membrane/membrane-dart-example_git.bb#L89) is 0.8.24 (which has the bugfix) but the log above shows 0.8.21 in line `/usr/src/debug/membrane-dart-example/git-r0/cargo_home/bitbake/serde_yaml-0.8.21/src/mapping.rs:16` which does not have the bugfix. Any idea why those should differ?

**Comment by jwinarske** [comment]
@jerel The recipe in Meta-flutter `kirkstone` builds fine.  I ran `cargo bitbake` on your tip of tree, which is where I hit the issue.  I was trying to pick up your recent changes before AGL roll.
